File: ios/App/DeepLinkRouter.swift
import UIKit
enum DeepLinkAction: String { case complete, miss, skip, prev, next }
final class DeepLinkRouter {
    static let scheme = "myapp"
    static func handle(url: URL) -> Bool {
        guard url.scheme == scheme, let action = DeepLinkAction(rawValue: url.host ?? "") else { return false }
        switch action {
        case .complete: StateStore.shared.mutate { $0.completeCurrent() }
        case .miss:     StateStore.shared.mutate { $0.missCurrent(applyPenalty: true) }
        case .skip:     StateStore.shared.mutate { $0.skipCurrent() }
        case .prev:     StateStore.shared.mutate { $0.prev() }
        case .next:     StateStore.shared.mutate { $0.next() }
        }
        WidgetBridge.requestReload(); return true
    }
}

File: android/app/src/main/java/com/petprogress/app/DeepLinkRouter.kt
package com.petprogress.app
import android.net.Uri
import android.content.Context
import com.petprogress.app.data.WidgetRepo
import com.petprogress.app.widget.updateAllWidgets
object DeepLinkRouter {
  fun handle(context: Context, uri: Uri): Boolean {
    val host = uri.host ?: return false
    val repo = WidgetRepo(context)
    when (host) {
      "complete" -> repo.mutate { it.completeCurrent() }
      "miss"     -> repo.mutate { it.missCurrent(applyPenalty = true) }
      "skip"     -> repo.mutate { it.skipCurrent() }
      "prev"     -> repo.mutate { it.prev() }
      "next"     -> repo.mutate { it.next() }
      else -> return false
    }
    updateAllWidgets(context); return true
  }
}